crates/sui-types/src/transaction.rs

```diff
@@ -2442,10 +2448,17 @@ impl InputObjects {
     }
 
     pub fn transaction_dependencies(&self) -> BTreeSet<TransactionDigest> {
-        self.objects
+        let mut dependencies: BTreeSet<TransactionDigest> = self
+            .objects
             .iter()
             .map(|(_, obj)| obj.previous_transaction)
-            .collect()
+            .collect();
+
+        for (_, _, digest) in &self.deleted {
+            dependencies.insert(*digest);
+        }
```

Comment:
nit: could use `.chain()` here instead of a second loop, but it doesn't really matter